### ISSUE TITLE
asar pack with electron fails on MacOS

### ISSUE THREAD

**Issue by Nitemaeric** [atom/asar#33]
So basically what I've found is this happens...

```
Error: ENOENT, no such file or directory '/Users/username/Development/project/node_modules/electron-prebuilt/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Frameworks'
```

So I looked up the file and it exists!

The path is wrong because there is a unescaped space between "Electron" and "Framework".

Adding a "\" should fix this, but I don't really want to dive into code that isn't mine.

**Comment by kyohei8** [comment]
+1 me too.
Electron version is `v0.25.3`

**Comment by minodisk** [comment]
+1 
same error in `v0.28.3`

**Comment by bompi88** [comment]
+1 mee too

**Comment by erikmellum** [comment]
+1 Me too!

Screenshot of the ENOENT error, and my ls -l showing that the file actually exists. The whole directory is wrapped in single quotes so even though the spaces are not escaped it should not matter.

http://prntscr.com/7qrmtg

**Comment by blmarket** [comment]
I tried logging nearby codes, and got this
```
{ files: {} }
{ files: {} }
{ files: {} }
{ files: {} }
{ files: { 'Electron Framework': { link: 'node_modules/electron-prebuilt/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework' } } }
{ link: 'node_modules/electron-prebuilt/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries' }
```

object contains link instead of files, so code should resolve path using that link.

I'll create PR for this.

**Comment by blmarket** [comment]
If you remove electron-prebuilt directory from node_modules, then it will work fine. (asar file does not need to contain electron-prebuilt)

It seems asar does not support properly for symbolic links. it tries to find entry for symlinked directory, and fails. :( I struggled to fix that, but soon I found I don't need to.

but if you have another dependency which contains symlink directory, you need to fix it though.

**Comment by ufo22940268** [comment]
@blmarket is right. You only need to remove `electron-prebuild` from node_modules of your project. Then build will succeed.

**Comment by FrankFang** [comment]
@blmarket is right.

**Comment by kurisubrooks** [comment]
Is there a way we can exclude certain files/folders from being packaged from the command line?

**Comment by kurisubrooks** [comment]
I removed `electron_prebuilt` from `node_modules`, but I still get this error.

**Comment by MathijsvVelde** [comment]
I use ```electron-rebuild```, it uses ```electron-prebuilt``` in it's ```node_modules```

**Comment by kevinsawicki** [comment]
Fixed by #48
